app_server/app/models/Storage.js

```diff
@@ -398,7 +398,53 @@ Class.getListOfStreams = function(originalFilters) {
 };
 
 /**
- * Return a list of streams from subscriptions
+ * Return a list of streams includes user, includes subscribers.
+ * Sorted with options.
+ * @param  {object} filters
+ * @param  {string} filters.sort
+ * @param  {string} filters.state
+ * @param  {object} filters.order
+ * @return {Promise<List<Sequelize.object>>} - a list of streams
+ */
+Class.getListOfStreamsForUser = function(originalFilters) {
+
+  var filters = mapParams(originalFilters);
+
+  if (filters.sort !== 'createdAt') {
+    return this.models.Stream.findAll({
+      include: [{
+        model: this.models.User,
+        as: 'streamer',
+        include: [{
+          model: this.models.User,
+          as: 'Subscribers'
+        }]
+      }],
+      where: {
+        live: filters.state
+      },
+      order: [[filters.sort, filters.order], ['createdAt', 'DESC']]
+    });
+  } else {
+    return this.models.Stream.findAll({
```

Comment:
This `else` clause is not necessary so you can make the code flatter here if you want to.